### DISCUSSION TITLE
Signature Solar 6500EX (MPP Clone)

### DISCUSSION THREAD

**Discussion by RoySalisbury** [syssi/esphome-pipsolar#34] [Q&A] [answered]
2 questions.  

1. Will this work with the 6500EX (MPP Clone) from Signature Solar?  It uses the same communications at all the other MPP solar inverters.
2. Since this inverter has a USB connection that basically allow you to plug in a USB to RS232 dongle (thats what I did with a Pi 4), I'm wondering if the ESP32 can just use the onboard USB port and communicate via serial that way.  I know when I plug it into my computer it just see's it as a serial port.

If this is possible, please point me in the direction of a .yaml config file for my ESP32.  I have all 8 of my JK BMS connected via bluetooth to ESP32 modules, and would love to get my 2 6500EX inverters connected as well.

Thanks.

**Comment by syssi** [comment, marked as answer]
1. I don't know. Does the device provide a RS232 port?
2. I'm not aware of an easy way to communicate via USB using ESPHome. You should use a Raspberry Pi and the `mppsolar` project instead if this is the only available communication port of your inverter.

**Reply by RoySalisbury** [reply]
This was a bit of an old post.  I had since found that the USB port on the inverter just exposes it as a HID device.  You can talk to it using the same protocol, but its a bit different.  Not sure of a way using an ESP32's USB port as a client instead of a host port.

**Reply by syssi** [reply]
Some ESPs (ESP32-S2) will come with a tiny USB stack and will support USB host mode. This would be the foundation to talk to USB-HID devices. For now this is far far away and you should stick to Mini PCs or Embedded Devices with Linux support if you want to talk to USB devices.

**Reply by SeByDocKy** [reply]
Really some ESP32-S2 could support USB-HID ? interesting .... for multiple JKBMS configuration :) with their own RS232 <=> USB converter.....

**Reply by syssi** [reply]
Lol. I will ping you in 2025 if I see a ESPHome node in USB host mode attached to a USB switch. ;-)

**Reply by SeByDocKy** [reply]
I am checking too :)
